internal/common/resource.go

```diff
@@ -72,6 +75,7 @@ type Resource struct {
 
 	// Generation configuration
 	ListIDsFunc                ResourceListIDsFunc
+	ListFunc                   ResourceListFunc
```

Comment:
I saw that you created a new value and a function to set it, but it isn't used, right?

Reply:
This is actually used [here](https://github.com/grafana/terraform-provider-grafana/blob/ca81e8d9440705a7e8c87352290815ae25ac7d15/internal/resources/cloud/resource_private_data_source_connect_network.go#L102).

It's not necessarily a requirement but I think it's a nice feature to be able to list all the resources of a specific type. If we don't want this then we can remove it 😊

Reply:
For example, `ListIDsFunc` is used [here](https://github.com/grafana/terraform-provider-grafana/blob/main/pkg/generate/generate.go#L207) and other parts for testing, but this new `LisFunc` is set but no one is calling it.

Reply:
That's fair, I'll remove this 😊